### DISCUSSION TITLE
Uploaded files are not shown in S3 Bucket

### DISCUSSION THREAD

**Discussion by yugalkishoregoyal** [brefphp/laravel-bridge#108] [Q&A] [answered]
Hi Guys,

I'm trying to upload the files to the S3 bucket using the below code. Code is running successfully and shows no errors, but when the file never gets uploaded to the bucket.

`
$attachment = $request->file('attachment');
$attachment->storePubliclyAs("/attachments/", $request->name, ['disk' => 's3']);
`

I tried this locally on my dev machine and the file uploaded without any issues. So there is some problem with the serverless config only. I'm using the below-mentioned versions -

laravel/framework - v9.52.5
bref/laravel-bridge - 2.1.0
league/flysystem-aws-s3-v3 - 3.0

Note - I had read somewhere that Lambda returns the response before it is finished uploading the file, not sure this is the case here.

Let me know if you need more information.

**Comment by georgeboot** [comment]
Can you show your serverless.yml? How have you configured your `s3` filesystem in Laravel?

**Comment by yugalkishoregoyal** [comment]
To configure the S3 filesystem, I simply added the credentials in the.env file and used the disk as S3 in the sample code provided in my question.

Please find the serverless.yml code below -
[serverless.yml.txt](https://github.com/brefphp/laravel-bridge/files/11151103/serverless.yml.txt)

**Reply by georgeboot** [reply]
Ah so you've deployed your `.env` file? Can you share that one, excluded the secrets of course?

**Reply by yugalkishoregoyal** [reply]
Please find the `.env` file attached.
[env.txt](https://github.com/brefphp/laravel-bridge/files/11156637/env.txt)

**Comment by mnapoli** [comment, marked as answer]
Could it be that the filesystem disk does not throw exceptions by default: https://laravel.com/docs/10.x/filesystem#failed-writes

I'm suspecting the credentials deployed in `.env` might not actually have the right permissions (or my not even be used by the app).

I'd recommend enabling `throw`, and not deploying custom credentials in `.env` but instead setting permissions on the function directly: https://www.serverless.com/framework/docs/providers/aws/guide/iam

**Reply by yugalkishoregoyal** [reply]
It works with the arn, many thanks !!
